src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java

```diff
@@ -619,11 +623,30 @@ synchronized void flush() throws InterruptedException, IOException, DeltaFailure
     commitOffset();
   }
 
+  private void directLoadToTarget(TableBlob blob) throws InterruptedException, DeltaFailureException {
```

Comment:
Actually, this can result in duplicate data if there is a failure during the snapshot processing and we reprocess events. To protect against this, we would have to write some state to remember if this is the first time, or wipe out the existing data if it already exists.

Reply:
Added state to indicate whether direct load was in progress per table. If such load was in progress and we receive CREATE_TABLE ddl again, it indicates that the snapshot was abandoned. Clearing table before proceeding in such cases.

Reply:
Are there use cases for writing to pre-existing tables? If there are, clearing the table may not be desired, though I'm not sure when you would ever want to snapshot to a pre-existing table.